### ISSUE TITLE
Make SRIOV operator independent of CNI

### ISSUE THREAD

**Issue by aojea** [k8snetworkplumbingwg/sriov-network-device-plugin#549]
## What would you like to be added?

Remove the dependency on CNI plugins

## What is the use case for this feature / enhancement?

Depending on CNI adds complexity, since SRIOV plugin already controls the whole lifecycle of the devices it can use CDI devices and its own APIs to handle the attachment of the additional devices to the Pods

I've presented and demod how to use CDIDevices to add network interfaces to pod in https://docs.google.com/presentation/d/1pjDCtpdbCSWaqCbBYWgzTxAewOVbMf6rUS5SbjAJAe8/edit#slide=id.g2c2d4de4b62_1_323

https://github.com/aojea/network-device-plugin

**Comment by SchSeba** [comment]
Hi @aojea, Thanks for this really interesting topic!

I check the presentation also video and the code.

there is only one point that I was not able to find about the deletion of the pod are we able to run some work in that case? for example change the internal name back/mac address/remove IP address and other stuff

**Comment by aojea** [comment]
Good question, let me figure that out

**Comment by aojea** [comment]
@SchSeba https://github.com/opencontainers/runtime-spec/blob/main/config.md#poststop

**Comment by SchSeba** [comment]
This is really nice!

will you be able to talk about it on our next community meeting?
you can find it in the k8smeet@gmail.com calendar its the "K8s Network & Resource Management WG Tech discussion" meeting

**Comment by zshi-redhat** [comment]
@aojea  Is the goal here to eliminate the use of sriov-cni or just handle the movement of sriov interface to container namespace (still use sriov-cni for interface configurations, such as IP/MAC/spoofCheck/linkAuto etc)?

**Comment by aojea** [comment]
> @aojea Is the goal here to eliminate the use of sriov-cni or just handle the movement of sriov interface to container namespace in sriov device plugin (still use sriov-cni for interface configurations, such as IP/MAC/spoofCheck/linkAuto etc)?

I'm looking for improvements on the user experience, not familiar with all the SRIOV use cases , but my feeling is that you can define this configurations in one place and avoid spreading the responsabilities over components this will turn out in a better UX and improve troubleshooting and supportability

**Comment by zshi-redhat** [comment]
> > @aojea Is the goal here to eliminate the use of sriov-cni or just handle the movement of sriov interface to container namespace in sriov device plugin (still use sriov-cni for interface configurations, such as IP/MAC/spoofCheck/linkAuto etc)?
> 
> I'm looking for improvements on the user experience and reduce the complexity of the existing solutions, not familiar with all the SRIOV use cases , but my feeling is that you can define this configurations in one place and avoid spreading the responsibilities over different components ... 

I assume it is in align with the network device proposal here: https://github.com/opencontainers/runtime-spec/issues/1239, which provides the capacity of adding a network device to the namespace without explicit or implicit interactions with CNI, but not to configure its networking parameters (e.g. IP allocation and management).

**Comment by aojea** [comment]
yep, exactly, decoupling interface management of interface configuration I think allow to have much cleaner implementations ... right now CNI is "abused" for "everything networking goes there"

**Comment by aojea** [comment]
updating the issue https://github.com/containerd/nri/pull/119

**Comment by aojea** [comment]
I think with DRA this has solved in a different layer
